public/pages/ClusterMetricsGraphs/components/ClusterMetricsGraphs.js

```diff
@@ -0,0 +1,208 @@
+  }
+
+  getXXYValues(data0, data) {
+    const x0 = data0.x;
+    const x = data.x;
+    const y = data0.y;
+    return { x0, x, y };
+  }
+
+  renderBarPlot = (data) => {
+    const xDomain_bar = [data[0].x0, data[data.length - 1].x];
+    const xTitle = 'Time';
+    return (
+      <ContentPanel title="Cluster Status" titleSize="s" panelStyles={{ paddingLeft: '10px' }}>
+        <FlexibleWidthXYPlot
+          height={100}
+          margin={{ top: 20, right: 20, bottom: 50 }}
+          xDomain={xDomain_bar}
+          xType="time"
+          yType="ordinal"
+          getColor={(dataPoint) => this.color_map[dataPoint.y]}
+          getY={() => 0}
+        >
+          <HorizontalBarSeries data={data} colorType="literal" />
+          <XAxis />
+          <ChartLabel text={xTitle} xPercent={0.965} yPercent={0.5} />
+        </FlexibleWidthXYPlot>
+      </ContentPanel>
+    );
+  };
+
+  render() {
+    const { dataPoints } = this.state;
+
+    if (!_.isEmpty(dataPoints)) {
+      const unassigned_shards_data = this.cleanData(dataPoints, 'unassigned_shards');
+      const cpu_data = this.cleanData(dataPoints, 'cpu_usage');
+
+      const min_jvm = this.cleanMinMaxData(dataPoints, 'jvm_pressure', 'minimum');
+      const max_jvm = this.cleanMinMaxData(dataPoints, 'jvm_pressure', 'maximum');
+
+      const min_cpu = this.cleanMinMaxData(dataPoints, 'cpu_usage', 'minimum');
+      const max_cpu = this.cleanMinMaxData(dataPoints, 'cpu_usage', 'maximum');
+
+      const jvm_pressure = this.cleanData(dataPoints, 'jvm_pressure');
+      const cluster_status_data = this.cleanData(dataPoints, 'cluster_status');
+      const cleaned_status = this.cleanBarData(cluster_status_data);
+      const number_pending_tasks = this.cleanData(dataPoints, 'number_of_pending_tasks');
+      const active_shards = this.cleanData(dataPoints, 'active_shards');
+      const relocating_shards = this.cleanData(dataPoints, 'relocating_shards');
+      const number_total_nodes = this.cleanData(dataPoints, 'number_of_nodes');
+      const number_data_nodes = this.cleanData(dataPoints, 'number_of_data_nodes');
+
+      return (
+        <>
+          {this.renderBarPlot(cleaned_status)}
+          <ClusterMetricsNode
+            data={number_total_nodes}
+            data2={number_data_nodes}
+            type={'Nodes Data'}
+          />
+          <ClusterMetricLineGraphMultiple
+            data={cpu_data}
+            dataMin={min_cpu}
+            dataMax={max_cpu}
+            type={'CPU Data'}
+          />
+          <ClusterMetricLineGraphMultiple
+            data={jvm_pressure}
+            dataMin={min_jvm}
+            dataMax={max_jvm}
+            type={'JVM Pressure'}
+          />
+          <ClusterMetricLineGraph data={unassigned_shards_data} type={'Unassigned Shards'} />
+          <ClusterMetricLineGraph data={active_shards} type={'Active Shards'} />
+          <ClusterMetricLineGraph data={relocating_shards} type={'Relocating Shards'} />
```

Comment:
I wonder if it would be more helpful to consolidate these 3 shard-related graphs into 1 like what you did for CPU and JVM with min, max, and avg.

Reply:
I had received feedback for having these graphs split up for readability. Not sure which would be a better idea.